src/features/views/components/ViewDataTable/columnTypes/PersonTagColumnType/index.tsx

```diff
@@ -0,0 +1,270 @@
+import { makeStyles } from '@mui/styles';
+import { useRouter } from 'next/router';
+import { Box, lighten, TextField, TextFieldProps } from '@mui/material';
+import { FC, KeyboardEvent, useCallback, useState } from 'react';
+import {
+  GridColDef,
+  GridRenderCellParams,
+  GridRenderEditCellParams,
+  MuiEvent,
+  useGridApiContext,
+} from '@mui/x-data-grid-pro';
+
+import compareTags from 'features/tags/utils/compareTags';
+import { DEFAULT_TAG_COLOR } from 'features/tags/components/TagManager/utils';
+import IApiClient from 'core/api/client/IApiClient';
+import { IColumnType } from '../';
+import { loadItemIfNecessary } from 'core/caching/cacheUtils';
+import TagChip from 'features/tags/components/TagManager/components/TagChip';
+import useAccessLevel from 'features/views/hooks/useAccessLevel';
+import useTag from 'features/tags/hooks/useTag';
+import useTagging from 'features/tags/hooks/useTagging';
+import { UseViewGridReturn } from 'features/views/hooks/useViewGrid';
+import ValueTagCell from './ValueTagCell';
+import { ZetkinObjectAccess } from 'core/api/types';
+import { ZetkinTag } from 'utils/types/zetkin';
+import ZUIFuture from 'zui/ZUIFuture';
+import { AppDispatch, RootState } from 'core/store';
+import { PersonTagViewColumn, ZetkinViewRow } from '../../../types';
+import { tagLoad, tagLoaded } from 'features/tags/store';
+
+type PersonTagViewCell = null | {
+  value?: string;
+};
+
+export default class PersonTagColumnType implements IColumnType {
+  cellToString(cell: PersonTagViewCell): string {
+    return cell?.value ? cell.value.toString() : Boolean(cell).toString();
+  }
+
+  getColDef(
+    column: PersonTagViewColumn,
+    accessLevel: ZetkinObjectAccess['level'],
+    state: RootState,
+    apiClient: IApiClient,
+    dispatch: AppDispatch,
+    orgId: number
+  ): Omit<GridColDef, 'field'> {
+    const tagId = column.config.tag_id;
+
+    const tagList = state.tags.tagList;
+    const tagItem = tagList.items.find((item) => item.id == tagId);
+
+    const tagFuture = loadItemIfNecessary(tagItem, dispatch, {
+      actionOnLoad: () => tagLoad(tagId),
+      actionOnSuccess: (tag) => tagLoaded(tag),
+      loader: () =>
+        apiClient.get<ZetkinTag>(`/api/orgs/${orgId}/people/tags/${tagId}`),
+    });
+
+    const tag = tagFuture.data;
+
+    return {
+      align: 'center',
+      editable: !accessLevel && tag?.value_type !== null,
+      headerAlign: 'center',
+      renderCell: (params: GridRenderCellParams<ZetkinViewRow, ZetkinTag>) => (
+        <Cell cellValue={params.value} personId={params.row.id} tag={tag} />
```

Comment:
Maybe this (combined with the other comments I've made above)?

```suggestion
        <Cell cellValue={params.value} personId={params.row.id} tag={tag || params.value} />
```